### DISCUSSION TITLE
Historische Daten auslesen

### DISCUSSION THREAD

**Discussion by SaschaHenning** [spali/go-rscp#14] [Q&A] [answered]
Ich stelle mich gerade etwas unfähig an und brauche bitte mal einen Tipp.

Aus dem E3DC möchte ich die historischen Daten auslesen, um im Home Assistant im Energieportal die Batterie-Lade und -Entladeleistung zu erfassen.

Dafür habe ich diesen Wiki Eintrag gefunden: https://github.com/spali/go-rscp/wiki/Example:-Request-for-historical-data

Leider bekomme ich es irgendwie nicht hin, dies von der Konsole aus abzurufen. 

`cat fileWithWikiContent.txt | jq --raw-input --slurp` (mit und ohne den Parametern) funktioniert nicht.

Danke für einen Tipp.

**Comment by spali** [comment, marked as answer]
Ich hatte es so im Einsatz:
```
jq -ncM "$(cat ./fileWithWikiContent.txt)"
```
Aber Achtung. Hatte festgestellt, dass mit den Zeitzonen irgend ein Bug drin ist. Mich störte der nicht, aber hatte in den Stunden um den Tageswechsel eine Stunde lang immer den gleichen Wert bekommen... Ich gehe stark davon aus, dass es irgendwo in der Zeitzonen Umrechnung noch ein Bug im JQ script drin ist.
Ich bin dazu übergegangen historischen Werte nicht direkt von E3DC zu nehmen. Ich nutze nun Home Assistant Funktionen dafür.

**Comment by maaque** [comment]
Ich versuche auch die Daten der laufenden Woche in das Script einzubauen, scheitere aber an der Syntax im Script. Kann mir eventuell jemand auf die Sprünge helfen wie die Zeilen 

`  # returns the start of the week of the given date as timestamp
  def startOfWeek: . | (
    . = if . == null then $now else . end |
    gmtime | [.[0],.[1],.[1],0,0,0,0,0] | mktime
  );
  
def daysInWeek: . | (
    . = if . == null then $now else . end |
    (. | gmtime | .[1] = 7 | mktime | . += timespanDay(4)) as $next_week |
    $next_week - timespanDay($next_week | gmtime | .[2]) | gmtime | .[2]
  );`

aussehen müssen?
